google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java

```diff
@@ -987,6 +987,19 @@ public static QueryResultsOption maxWaitTime(long maxWaitTime) {
   QueryResponse query(QueryRequest request);
```

Comment:
Are we open to making breaking changes here? I'd somewhat like to do away with QueryRequest/QueryResponse altogether if possible and return a job here, especially since Job has `waitFor` but QueryResponse doesn't.

We tell users to run their own wait loop for QueryResponse

```
 while (!response.jobCompleted()) {
   Thread.sleep(1000);
   response = bigquery.getQueryResults(response.getJobId());
 }
```

We may need a special QueryJob class though so that query jobs can poll for completion using `getQueryResults` per BQ team's go/bq-client-reqs requirements doc.